<Type Name="ServerException" FullName="System.Runtime.Remoting.ServerException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="1e5a3b3fea44356298c675f34231de1b26aaecbe" /><Meta Name="ms.sourcegitcommit" Value="11d168140aa8fade0768c2a9dde3e3bcacfdfb7d" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="ja-JP" /><Meta Name="ms.lasthandoff" Value="05/27/2020" /><Meta Name="ms.locfileid" Value="83955547" /></Metadata><TypeSignature Language="C#" Value="public class ServerException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit ServerException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.Runtime.Remoting.ServerException" />
  <TypeSignature Language="VB.NET" Value="Public Class ServerException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class ServerException : SystemException" />
  <TypeSignature Language="F#" Value="type ServerException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute>
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="36294-101">クライアントが、例外をスローできない非 .NET Framework アプリケーションに接続する場合に、クライアントにエラーを通知するためにスローされる例外。</span><span class="sxs-lookup"><span data-stu-id="36294-101">The exception that is thrown to communicate errors to the client when the client connects to non-.NET Framework applications that cannot throw exceptions.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="36294-102"><xref:System.Runtime.Remoting.ServerException> では、値0x8013150E を持つ HRESULT COR_E_SERVER を使用します。</span><span class="sxs-lookup"><span data-stu-id="36294-102"><xref:System.Runtime.Remoting.ServerException> uses the HRESULT COR_E_SERVER, which has the value 0x8013150E.</span></span>  
  
 <span data-ttu-id="36294-103"><xref:System.Runtime.Remoting.ServerException> では、参照の等価性をサポートする既定の <xref:System.Object.Equals%2A> 実装が使用されます。</span><span class="sxs-lookup"><span data-stu-id="36294-103"><xref:System.Runtime.Remoting.ServerException> uses the default <xref:System.Object.Equals%2A> implementation, which supports reference equality.</span></span>  
  
 <span data-ttu-id="36294-104">インスタンスの初期プロパティ値の一覧については<xref:System.Runtime.Remoting.ServerException>を参照してください、<xref:System.Runtime.Remoting.ServerException>コンス トラクター。</span><span class="sxs-lookup"><span data-stu-id="36294-104">For a list of initial property values for an instance of <xref:System.Runtime.Remoting.ServerException>, see the <xref:System.Runtime.Remoting.ServerException> constructors.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
    <related type="Article" href="/dotnet/standard/exceptions/"><span data-ttu-id="36294-105">.NET での例外の処理とスロー</span><span class="sxs-lookup"><span data-stu-id="36294-105">Handling and throwing exceptions in .NET</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="36294-106"><see cref="T:System.Runtime.Remoting.ServerException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="36294-106">Initializes a new instance of the <see cref="T:System.Runtime.Remoting.ServerException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ServerException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.Remoting.ServerException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ServerException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="36294-107"><see cref="T:System.Runtime.Remoting.ServerException" /> クラスの新しいインスタンスを、既定のプロパティを使用して初期化します。</span><span class="sxs-lookup"><span data-stu-id="36294-107">Initializes a new instance of the <see cref="T:System.Runtime.Remoting.ServerException" /> class with default properties.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="36294-108"><xref:System.Runtime.Remoting.ServerException> は <xref:System.SystemException> クラスから継承されます。</span><span class="sxs-lookup"><span data-stu-id="36294-108"><xref:System.Runtime.Remoting.ServerException> inherits from the <xref:System.SystemException> class.</span></span> <span data-ttu-id="36294-109">次の表に、<xref:System.Runtime.Remoting.ServerException> クラスのインスタンスのプロパティの初期値を示します。</span><span class="sxs-lookup"><span data-stu-id="36294-109">The following table shows initial property values for an instance of the <xref:System.Runtime.Remoting.ServerException> class:</span></span>  
  
|<span data-ttu-id="36294-110">プロパティ</span><span class="sxs-lookup"><span data-stu-id="36294-110">Property</span></span>|<span data-ttu-id="36294-111">値</span><span class="sxs-lookup"><span data-stu-id="36294-111">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|`null`|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="36294-112">空の文字列 ("")</span><span class="sxs-lookup"><span data-stu-id="36294-112">The empty string ("")</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ServerException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.Remoting.ServerException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ServerException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Runtime.Remoting.ServerException : string -&gt; System.Runtime.Remoting.ServerException" Usage="new System.Runtime.Remoting.ServerException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="36294-113">例外を説明するメッセージ。</span><span class="sxs-lookup"><span data-stu-id="36294-113">The message that describes the exception</span></span></param>
        <summary><span data-ttu-id="36294-114">指定したメッセージを使用して、<see cref="T:System.Runtime.Remoting.ServerException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="36294-114">Initializes a new instance of the <see cref="T:System.Runtime.Remoting.ServerException" /> class with a specified message.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="36294-115"><xref:System.Runtime.Remoting.ServerException> は <xref:System.SystemException> クラスから継承されます。</span><span class="sxs-lookup"><span data-stu-id="36294-115"><xref:System.Runtime.Remoting.ServerException> inherits from the <xref:System.SystemException> class.</span></span> <span data-ttu-id="36294-116">次の表に、<xref:System.Runtime.Remoting.ServerException> クラスのインスタンスのプロパティの初期値を示します。</span><span class="sxs-lookup"><span data-stu-id="36294-116">The following table shows initial property values for an instance of the <xref:System.Runtime.Remoting.ServerException> class:</span></span>  
  
|<span data-ttu-id="36294-117">プロパティの型</span><span class="sxs-lookup"><span data-stu-id="36294-117">Property Type</span></span>|<span data-ttu-id="36294-118">条件</span><span class="sxs-lookup"><span data-stu-id="36294-118">Condition</span></span>|  
|-------------------|---------------|  
|<xref:System.Exception.InnerException%2A>|`null`|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="36294-119">`message` 文字列</span><span class="sxs-lookup"><span data-stu-id="36294-119">The `message` string</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ServerException (string message, Exception InnerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception InnerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.Remoting.ServerException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, InnerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ServerException(System::String ^ message, Exception ^ InnerException);" />
      <MemberSignature Language="F#" Value="new System.Runtime.Remoting.ServerException : string * Exception -&gt; System.Runtime.Remoting.ServerException" Usage="new System.Runtime.Remoting.ServerException (message, InnerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="InnerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="36294-120">例外の原因を示すエラー メッセージです。</span><span class="sxs-lookup"><span data-stu-id="36294-120">The error message that explains the reason for the exception.</span></span></param>
        <param name="InnerException"><span data-ttu-id="36294-121">現在の例外の原因になった例外です。</span><span class="sxs-lookup"><span data-stu-id="36294-121">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="36294-122"><paramref name="InnerException" /> パラメーターが <see langword="null" /> でない場合は、内部例外を処理する <see langword="catch" /> ブロックで現在の例外が発生します。</span><span class="sxs-lookup"><span data-stu-id="36294-122">If the <paramref name="InnerException" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="36294-123">指定したエラー メッセージおよびこの例外の原因となった内部例外への参照を使用して、<see cref="T:System.Runtime.Remoting.ServerException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="36294-123">Initializes a new instance of the <see cref="T:System.Runtime.Remoting.ServerException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="36294-124">前回の例外の直接的な結果としてスローされる例外は、<xref:System.Exception.InnerException%2A> プロパティに前回の例外への参照を含める必要があります。</span><span class="sxs-lookup"><span data-stu-id="36294-124">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="36294-125"><xref:System.Exception.InnerException%2A> プロパティは、コンストラクターに渡されるのと同じ値を返します。または、<xref:System.Exception.InnerException%2A> プロパティがコンストラクターに内部例外値を提供しない場合には `null` を返します。</span><span class="sxs-lookup"><span data-stu-id="36294-125">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="36294-126">次の表に、<xref:System.Runtime.Remoting.ServerException> クラスのインスタンスのプロパティの初期値を示します。</span><span class="sxs-lookup"><span data-stu-id="36294-126">The following table shows initial property values for an instance of the <xref:System.Runtime.Remoting.ServerException> class:</span></span>  
  
|<span data-ttu-id="36294-127">プロパティ</span><span class="sxs-lookup"><span data-stu-id="36294-127">Property</span></span>|<span data-ttu-id="36294-128">値</span><span class="sxs-lookup"><span data-stu-id="36294-128">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="36294-129">内部例外参照</span><span class="sxs-lookup"><span data-stu-id="36294-129">The inner exception reference</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="36294-130">エラーメッセージ文字列</span><span class="sxs-lookup"><span data-stu-id="36294-130">The error message string</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
        <related type="Article" href="/dotnet/standard/exceptions/"><span data-ttu-id="36294-131">.NET での例外の処理とスロー</span><span class="sxs-lookup"><span data-stu-id="36294-131">Handling and throwing exceptions in .NET</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>
